### PR TITLE
Rob Youngman's changes so far

### PR DESCRIPTION
**Pull request by youngmanr**

### CODE REVIEW

**Comment by tansaku** [comment]
You had your chitter_challenge reviewed by **jamie**.
### The good points are:
* Correct installation instructions in README
* All unnecessary components removed, Ruby style guide conventions followed
* Rakefile has appropriate tasks
* Gemfile uses test groups
* spec_helper.rb pulls in entire app through single point, Only test helpers (not view helpers) are pulled in to spec_helper.rb
* Database Cleaner is set up correctly
* DataMapper set up in separate configuration file
* Feature tests organized correctly
* Spec 'Helpers' used appropriately
* All controller methods short and succinct
* Forms use correct markup
* Correct use of Semantic HTML elements

### You should consider the following improvements:
* Ensure Correct Location of Feature/Acceptance and Unit Tests
* Ensure Asset Routes are Set Correctly
* Prefer Modular over Classic Style
* Use flash.now and flash.next Correctly
* Split Routes into Separate Controller files
* Use Partials with Standard HTML Conventions
* Use View Helpers

see https://github.com/makersacademy/chitter_challenge/blob/master/docs/review.md for more details

File: spec/features/user_log_in_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+                password: 'secret1234',
+                password_confirmation: 'secret1234',
+                name: 'user',
+                user_name: 'user1')
+  end
+
+  scenario 'with correct credentials' do
+    sign_in(email: user.email, password: user.password)
+    expect(page).to have_content "Welcome, #{user.email}"
+  end
```
Comment (danldb) [inline spec/features/user_log_in_spec.rb]: good sign_in method - the argument makes it really clear that you should be seeing the users email.

File: app/app.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      erb :'peeps/new'
+    else
+      flash.next[:notice] = 'User not logged in'
+      redirect '/peeps'
+    end
+  end
+
+  post '/peeps' do
+    Peep.create(message: params[:message], name: current_user.name,
+      user_name: current_user.user_name, time_stamp: Time.now)
```
Comment (danldb) [inline app/app.rb]: what happens if the user later changes their username?

**Comment by danldb** [comment]
:checkered_flag: 

Really good submission, I don't have a lot to add. Good separation between view / model / controller. If I was being really picky I could mention that the controllers could be refactored into separate files, but the app is really small so even that might be overkill.. Well done!

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  def self.authenticate(email, password)
+    user = first(email: email)
+    if user && BCrypt::Password.new(user.password_digest) == password
+      user
+    else
+      nil
+    end
+  end
+end
```
Comment (danldb) [inline app/models/user.rb]: no relationship between user and peeps? How would you get all the peeps that belong to a user?

**Pull request closed without merging**